WordPress/gutenberg
# env: Add Support for "afterStop" Lifecycle Script

## What?


The `@wordpress/env` Package already supports the following Lifecycle Scripts:

* `afterClean`
* `afterDestroy`
* `afterStart`

This is currently documented at:

* https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/#lifecycle-scripts
* https://github.com/WordPress/gutenberg/blob/ddadd1a95d18270908ac4a1fd8d6e354cfadf61c/packages/env/README.md#lifecycle-scripts

This PR adds an additional opportunity (`afterStop`) for users to invoke a script after the environment is stopped.

## Why?



This script gives the ability to clean up resources that were spun up via the `afterStart` lifecycle script.

## How?



Use the same mechanisms as the existing lifecycle scripts:

* Attach the `scripts` argument in `packages/env/lib/cli.js` with the default of "true"
* Pass that argument through to the stop script
* If it's true, run any configured scripts in the top-level `lifecycleScripts.afterStop` portion of `.wp-env.json`

## Testing Instructions

* Create a `.wp-env.json` with contents:
    ```
    {
        "lifecycleScripts": {
          "afterClean": "echo 'All clean!' > /tmp/wpenv.log",
          "afterDestroy": "echo 'Destroyed!' > /tmp/wpenv.log",
          "afterStart": "echo 'WordPress is ready!' > /tmp/wpenv.log",
          "afterStop": "echo 'Goodbye!' > /tmp/wpenv.log"
        }
    }
    ```
* In a dedicated terminal, run `tail -F /tmp/wpenv.log`
* Run `wp-env start`
    * You should see a new log line: `WordPress is ready!`
* Repeat with the `clean`, `destroy`, and `stop` subcommands.
    * You should see corresponding log lines for each.
* Automated tests should pass in this PR
